File: Lighthouse.Frontend/src/models/LighthouseRelease/LighthouseRelease.ts
import { ILighthouseReleaseAsset } from "./LighthouseReleaseAsset";

export interface ILighthouseRelease {
    name: string;
    link: string;
    highlights: string;
    version: string;
    assets: ILighthouseReleaseAsset[];
}

export class LighthouseRelease implements ILighthouseRelease {
    name: string;
    link: string;
    highlights: string;
    version: string;
    assets: ILighthouseReleaseAsset[];

    constructor(name: string, link: string, highlights: string, version: string, assets: ILighthouseReleaseAsset[]) {
        this.name = name;
        this.link = link;
        this.highlights = highlights;
        this.version = version;
        this.assets = assets;
    }
}

File: Lighthouse.Backend/Lighthouse.Backend/Models/LighthouseRelease.cs
namespace Lighthouse.Backend.Models
{
    public class LighthouseRelease
    {
        public string Name { get; set; }

        public string Link { get; set; }

        public string Highlights { get; set; }

        public string Version { get; set; }

        public List<LighthouseReleaseAsset> Assets { get; } = [];
    }
}
